### PR TITLE
[pyspark] Implement SparkXGBRanker estimator

### PR DESCRIPTION
**Pull request by WeichenXu123**
Closes https://github.com/dmlc/xgboost/issues/8110

Implement SparkXGBRanker estimator.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @trivialfis @wbo4958  Thank you !

**Comment by trivialfis** [comment]
Apologies for the delay, will take a deeper look tomorrow.

**Comment by WeichenXu123** [comment]
CC @trivialfis @wbo4958 Thanks!

**Review by trivialfis — APPROVED** [review]

File: python-package/xgboost/spark/estimator.py [hunk 1]
```diff
@@ -213,5 +221,126 @@ def _xgb_cls(cls):
...
+                "Spark Xgboost classifier estimator does not support `qid_col` param."
+            )
+
 
 _set_pyspark_xgb_cls_param_attrs(SparkXGBClassifier, SparkXGBClassifierModel)
+
+
+class SparkXGBRanker(_SparkXGBEstimator):
+    """SparkXGBRanker is a PySpark ML estimator. It implements the XGBoost
+    classification algorithm based on XGBoost python library, and it can be used in
```
[the remark this answers is not in the document]
Comment (trivialfis) [reply python-package/xgboost/spark/estimator.py:237]: @WeichenXu123 Would you like to open a fix?

**Pull request merged**